### PR TITLE
Add faster blob range read implementation

### PR DESCRIPTION
**Pull request by lgruen**
See #21 for context.

```bash
$ time gsutil cat -r 65536-98303 gs://cpg-seqr-testing/NA12878_trio/NA12892.cram > /dev/null
0.86s user 0.42s system 57% cpu 2.223 total
```

Fetching the blob range using the Python client library takes about ~0.6s.

### CODE REVIEW

**Review by illusional — APPROVED** [review]
Ah this is cool!

**Comment by illusional** [comment]
Unit tests are failing BTW:

> google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see

**Comment by illusional** [comment]
Still no dice, here's the full stack:

```
======================================================================
ERROR: test_proxy_google_to_igv (seqr.views.apis.igv_api_tests.IgvAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/mock/mock.py", line 1346, in patched
    return func(*newargs, **newkeywargs)
  File "/home/runner/work/seqr/seqr/seqr/views/apis/igv_api_tests.py", line 27, in test_proxy_google_to_igv
    self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT)
  File "/home/runner/work/seqr/seqr/seqr/views/apis/igv_api_tests.py", line 27, in <listcomp>
    self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/django/test/client.py", line 89, in closing_iterator_wrapper
    yield from iterable
  File "/home/runner/work/seqr/seqr/seqr/utils/file_utils.py", line 46, in file_iter
    for line in _google_bucket_file_iter(file_path, byte_range=byte_range, raw_content=raw_content):
  File "/home/runner/work/seqr/seqr/seqr/utils/file_utils.py", line 69, in _google_bucket_file_iter
    bucket = gcs_storage_client().bucket(path_segments[2])
  File "/home/runner/work/seqr/seqr/seqr/utils/file_utils.py", line 16, in gcs_storage_client
    storage_client = storage.Client()
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/cloud/storage/client.py", line 126, in __init__
    _http=_http,
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/cloud/client.py", line 277, in __init__
    _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/cloud/client.py", line 225, in __init__
    project = self._determine_default(project)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/cloud/client.py", line 244, in _determine_default
    return _determine_default_project(project)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/cloud/_helpers.py", line 186, in _determine_default_project
    _, project = google.auth.default()
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/google/auth/_default.py", line 483, in default
    raise exceptions.DefaultCredentialsError(_HELP_MESSAGE)
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
```

**Comment by lgruen** [comment]
This is now ready for another look.

Please squash-merge :).

File: seqr/utils/file_utils.py [hunk 1]
```diff
@@ -32,8 +45,23 @@ def does_file_exist(file_path):
 
 def file_iter(file_path, byte_range=None, raw_content=False):
     if _is_google_bucket_file_path(file_path):
-        for line in _google_bucket_file_iter(file_path, byte_range=byte_range, raw_content=raw_content):
-            yield line
+        path_segments = file_path.split('/')
+        if len(path_segments) < 4:
```
Comment (illusional) [inline seqr/utils/file_utils.py:49]: Why 4 path segments?

**Review by illusional — APPROVED** [review]
Looks great!

File: seqr/utils/file_utils.py (hunk 1 above)
Comment (lgruen) [reply seqr/utils/file_utils.py:49]: It's because of the double slash after `gs:`:
1. `'gs:'`
2. `''` (empty)
3. `<bucket_name>`
4. `<path>`

**Comment by lgruen** [comment]
> Looks great, if this works, should we upstream it?

Yes, just want to verify it actually makes a difference :)

**Pull request merged**